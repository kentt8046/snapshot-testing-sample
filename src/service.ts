import { pseudoRandomBytes } from "crypto";

interface User {
  firstName: string;
  lastName: string;
  email: string;
  accountName: string;
}

const userStore = new Map<string, User>();
export function createUser(user: User): User & { id: string } {
  const id = pseudoRandomBytes(64).toString("hex");
  const entity = { id, ...user };
  userStore.set(id, entity)
  return entity;
}
