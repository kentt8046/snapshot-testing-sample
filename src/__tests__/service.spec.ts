import "jest";

import * as service from "../service";

describe("service", () => {

  it("createUser", () => {
    const input = {
      firstName: "太郎",
      lastName: "てすと",
      email: "test@email.com",
      accountName: "test",
    };
    const user = service.createUser(input);
    expect(user).toMatchSnapshot();
  });

});
