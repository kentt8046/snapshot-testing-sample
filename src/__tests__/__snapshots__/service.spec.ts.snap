// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service createUser 1`] = `
Object {
  "accountName": "test",
  "email": "test@email.com",
  "firstName": "太郎",
  "id": "3fdce6210aa07fbc418b5f03a4479b576f9d4c5c2c6d74f245e4d35e35e77a1efe1469f44ce123e2dd94ba4336f8ad2673120cf97d03ae2f6cdf1935133cd788",
  "lastName": "てすと",
}
`;
